### PR TITLE
Make hpack micro bm client initial metadata more representative

### PR DESCRIPTION
**Pull request by yang-g**
- Make :path element not indexed as it is the current status
- Add two binary headers (not indexed)
- Fixed label printing

This will make the micro-benchmarks take longer.
On my workstation, encoding 215ns --> 455ns, decoding 186ns --> 761ns.

### CODE REVIEW

File: test/cpp/microbenchmarks/bm_chttp2_hpack.cc [hunk 1]
```diff
@@ -545,30 +603,51 @@ class NonIndexedBinaryElem<100, false> {
...
-        "@\x07:scheme\x04http"
-        "@\x07:method\x04POST"
-        "@\x0a:authority\x09localhost"
-        "@\x0c"
-        "content-type\x10"
-        "application/grpc"
-        "@\x14grpc-accept-encoding\x15identity,deflate,gzip"
-        "@\x02te\x08trailers"
-        "@\x0auser-agent\"bad-client grpc-c/0.12.0.0 (linux)")};
+    return {MakeSlice(
```
Comment (ctiller) [inline test/cpp/microbenchmarks/bm_chttp2_hpack.cc]: Can we retain the existing thing also, and add a new case (RepresentativeClientInitialMetadata2... or some better name)

File: test/cpp/microbenchmarks/bm_chttp2_hpack.cc [hunk 2]
```diff
@@ -206,10 +257,24 @@ class RepresentativeClientInitialMetadata {
         GRPC_MDELEM_SCHEME_HTTP, GRPC_MDELEM_METHOD_POST,
         grpc_mdelem_from_slices(
             exec_ctx, GRPC_MDSTR_PATH,
-            grpc_slice_intern(grpc_slice_from_static_string("/foo/bar"))),
+            grpc_slice_from_static_string("/grpc.test.FooService/BarMethod")),
         grpc_mdelem_from_slices(exec_ctx, GRPC_MDSTR_AUTHORITY,
                                 grpc_slice_intern(grpc_slice_from_static_string(
                                     "foo.test.google.fr:1234"))),
+        grpc_mdelem_from_slices(
```
Comment (ctiller) [inline test/cpp/microbenchmarks/bm_chttp2_hpack.cc]: Maybe break this out as a separate fixture also

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

File: test/cpp/microbenchmarks/bm_chttp2_hpack.cc (hunk 2 above)
Comment (yang-g) [reply test/cpp/microbenchmarks/bm_chttp2_hpack.cc]: Restored the existing fixtures and made the new stuff new fixtures. PTAL

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] Performance differences noted:
Benchmark                          cpu_time    real_time
---------------------------------  ----------  -----------
BM_ErrorGetIntFromNoError          +7%         +7%
BM_ErrorGetStatusCode<ErrorNone>   +5%         +5%
BM_ErrorHttpError<ErrorCancelled>  +7%         +7%
BM_ErrorHttpError<ErrorNone>       +6%         +6%
BM_MetadataRefUnrefStatic          -4%         -4%
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] Performance differences noted:
Benchmark                                    cpu_time    real_time
-------------------------------------------  ----------  -----------
BM_ErrorGetIntFromNoError                    +4%         +4%
BM_ErrorGetStatus<ErrorWithGrpcStatus>       +8%         +8%
BM_ErrorGetStatus<ErrorWithHttpError>        +5%         +5%
BM_MetadataRefUnrefExternal                  +5%         +5%
BM_MetadataRefUnrefStatic                    -16%        -16%
BM_PumpStreamServerToClient<InProcess>/256k  -5%         -5%
```

**Comment by yang-g** [comment]
Flakes: #10763 #12618 #12510

**Comment by yang-g** [comment]
This will be merged to #12946. Closing.

**Pull request closed without merging**